### DISCUSSION TITLE
I made an all-in-one script to download, translate and subtitle any video from Youtube, Twitter etc.

### DISCUSSION THREAD

**Discussion by danirukun** [openai/whisper#387] [Show and tell]
I made a small script to use the [CPP Whisper implementation](https://github.com/openai/whisper/discussions/130) that also uses the models, with `ffmpeg` and `yt-dlp` under the hood to make the whole process easier:

https://gist.github.com/DaniruKun/96f763ec1a037cc92fe1a059b643b818

Basically it:
1. Downloads the video
2. Extracts the audio track and resamples it for whisper
3. Does the transcribe and generated subtitle file
4. Embeds the subtitle track into the source media file

It is as simple as

```shell
WHISPER_LANG=ja MODEL_PATH=models/ggml-medium.bin ./whisper-transcribe.bash https://www.youtube.com/watch\?v\=TRrYyGh6FsM
```

**Comment by claudchereji** [comment]
Yo I’ve been looking for something like this for podcasts. Would it work for xml feeds? Grabbing the mp3 file and doing the same as it would for the YouTube audio file? Also, does whisper do timestamps?

**Reply by abi** [reply]
Yes, Whisper does do timestamps. You could make this grab an MP3 file and transcribe it.

**Reply by aafiyahtech** [reply]
Hi, even I was trying to work on mp3 or other audio formats. would love to know your comments on how should I go about it
